src/main/java/org/jenkinsci/maven/plugins/hpi/ValidateMojo.java

```diff
@@ -15,6 +17,20 @@
 @Mojo(name = "validate", defaultPhase = LifecyclePhase.VALIDATE)
 public class ValidateMojo extends AbstractJenkinsMojo {
 
+    private static final String HTTP_GITHUB_COM = "http://github.com/";
+    private static final String HTTPS_GITHUB_COM = "https://github.com/";
+
+    private static final String SCM_GIT_GIT_URL_BAD = "scm:git:git://github.com/";
+    private static final String SCM_GIT_HTTP_URL_BAD = "scm:git:" + HTTP_GITHUB_COM;
+    private static final String SCM_GIT_SSH_URL_BAD = "scm:git:ssh://git@github.com/";
+
+    private static final String SCM_GIT_HTTPS_URL_GOOD = "scm:git:" + HTTPS_GITHUB_COM;
+    private static final String SCM_GIT_SSH_URL_GOOD = "scm:git:git@github.com:";
+
+    private static final String GIT_URLS_ARE_DEPRECATED = "git:// URLs are deprecated";
+    private static final String HTTP_URLS_ARE_INSECURE = "http:// URLs are insecure";
+    private static final String SSH_URLS_DO_NOT_WORK_WELL_WITH_PCT = "ssh:// URLs do not work well with PCT";
```

Comment:
I don't think it's really an argument. If they don't work well with pct as is, PCT should run with something like

```
[url "https://github.com/"]
	insteadOf = "git@github.com:"
	pushInsteadOf = "git@github.com:"
```

in its git configuration, guaranteeing that https is used in a homogeneous way even if the plugin has ssh configured.

Other variants can be used for other protocols.